internal_ws/ykpack/src/types.rs

```diff
@@ -492,6 +492,10 @@ pub enum Statement {
     /// Cast a value into another. Since the cast type and the destination type are the same, we
     /// only need the latter.
     Cast(IRPlace, IRPlace),
+    /// Marks the spot where we want trace stitching to start.
+    LoopStart,
+    /// Jumps to the trace stitching marker.
+    LoopEnd,
```

Comment:
Is there any scenario where we can have >1 `LoopEnd` I wonder?

Reply:
I don't think so. That would imply that there are multiple returns in a single trace, which can't happen since by definition a trace only contains a single path.

Reply:
> That would imply that there are multiple returns in a single trace, which can't happen since by definition a trace only contains a single path.

Once we have side traces, we will have a tree of traces, and then in a sense one does end up with multiple loop ends.

Reply:
I guess from that perspective that's true, side-traces also reintroduce control-flow, making it possible to have multiple returns.

Reply:
At that point we can revise our bytecode names.

Reply:
If my memory serves me well, then we've briefly discussed at some point, that we'll need to give `LoopStart` an id, so we can jump to it from another trace. But I think we decided to leave that for a future PR unless there are strong reasons why we should do it now.

Reply:
It's not just the names I'm worried about, it's whether they embed other assumptions that we might soon invalidate, and that it might be easiest to `todo!()` now.

Reply:
I'm not sure what kind of `todo` you expect here. The only thing we can do here is give LoopStart/LoopEnd an id parameter so we can distinguish between different loops.

Reply:
Ah, I think I understand. OK let's not worry about this for the time being.